process/transaction/baseProcess.go

```diff
@@ -152,10 +152,20 @@ func (txProc *baseTxProcessor) checkTxValues(
 		txFee = txProc.economicsFee.ComputeTxFee(tx)
 	}
 
-	if acntSnd.GetBalance().Cmp(txFee) < 0 {
+	feePayer, isRelayedV3, err := txProc.getFeePayer(tx, acntSnd)
+	if err != nil {
+		return err
+	}
+
+	// early exit for relayed v3, the cost will be compared with the sender balance
+	if isRelayedV3 {
```

Comment:
it should be few lines later, indeed

the reason why it is needed is because this method is called from multiple places:
1. for full relayed tx, before moving the funds from relayer to the sender. In this case, we need the early exit as the feePayer will be the relayer, but the txValue will be taken from the sender. So we don't want to check if the relayer also has the tx.Value.
2. only for user tx, when the tx fee was already moved into the relayer account. At this point, the sender of the user tx will be verified if it has both the fee(moved from relayer) and the tx.Value (which is here defined as `cost`)

Reply:
it is too hard to understand the code and a lot of mistakes can happen because of it.

Can you have separate functions called as it should be ?